src/runtime/slots/mp_length.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Reflection;
+
+namespace Python.Runtime.Slots
+{
+    static class mp_length_slot
+    {
+        /// <summary>
+        /// Implements __len__ for classes that implement ICollection
+        /// (this includes any IList implementer or Array subclass)
+        /// </summary>
+        public static int mp_length(IntPtr ob)
+        {
+            var co = ManagedType.GetManagedObject(ob) as CLRObject;
+            if (co == null)
+            {
+                Exceptions.RaiseTypeError("invalid object");
+            }
+
+            Type t = co.inst.GetType();
+            PropertyInfo p = t.GetProperty("Count");
```

Comment:
```csharp
> typeof(int[]).GetProperty("Count")
null
```

E.g. you need to look for explicit interface implementations.
An option is to loop through the implemented interfaces as you did in `CreateType`, and then get `Count` property from it instead.

Reply:
`typeof(ICollection<int>).GetProperty("Count").GetValue(new int[10])`